### PR TITLE
Show "update your connection" in User repositories view 

### PR DESCRIPTION
**Pull request by rafax**
Show banner from CLOUD-316 in user owned repos too.

## Test plan

- tested locally
- https://www.loom.com/share/31e369f1853a4954b48e541786bd66fb
## App preview:
- [Link](https://sg-web-cloud-317.onrender.com)

### CODE REVIEW

File: cmd/frontend/graphqlbackend/user.go [hunk 1]
```diff
@@ -535,6 +541,76 @@ func (r *UserResolver) BatchChangesCodeHosts(ctx context.Context, args *ListBatc
...
+	if !envvar.SourcegraphDotComMode() {
+		return false
+
+	}
+
+	overrides, err := db.FeatureFlags().GetOrgOverridesForUser(ctx, userID)
+	if err != nil {
+		return false
+	}
+	potentialUpgradeOrgs := map[int32]struct{}{}
```
Comment (rafax) [inline cmd/frontend/graphqlbackend/user.go:557]: @pjlast is this logic ok for now? For when we launch Github App?

File: cmd/frontend/graphqlbackend/user.go [hunk 2]
```diff
@@ -535,6 +541,76 @@ func (r *UserResolver) BatchChangesCodeHosts(ctx context.Context, args *ListBatc
...
+	for _, o := range orgs {
+		if _, ok := potentialUpgradeOrgs[o.OrgID]; ok {
+			queryFor = append(queryFor, o.OrgID)
+		}
+	}
+	if len(queryFor) == 0 {
+		return false
+	}
+	orgServices := []*types.ExternalService{}
+	for _, orgID := range queryFor {
```
Comment (rafax) [inline cmd/frontend/graphqlbackend/user.go:585]: This exibits "Select N+1" problem - I will work on refactoring ExternalServicesStore in separate PR (a lot of repetetive changes required).

**Comment by miveronese** [comment]
> @pjlast what's the latest approach to Github App flag -> should I drop it from backend? Should I try to detect orgs that did not upgrade to Github App yet?

@rafax , about the GithubApp flag, I have this PR that removes it from both backend and frontend: https://github.com/sourcegraph/sourcegraph/pull/33370/files

File: cmd/frontend/graphqlbackend/user.go [hunk 3]
```diff
@@ -535,6 +541,76 @@ func (r *UserResolver) BatchChangesCodeHosts(ctx context.Context, args *ListBatc
...
+
+	}
+
+	overrides, err := db.FeatureFlags().GetOrgOverridesForUser(ctx, userID)
+	if err != nil {
+		return false
+	}
+	potentialUpgradeOrgs := map[int32]struct{}{}
+	for _, override := range overrides {
+		if override.FlagName == "github-app-cloud" {
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/user.go:559]: cc @miveronese a new usage of GitHub App feature flag to be taken care by your PR.

File: cmd/frontend/graphqlbackend/user.go [hunk 4]
```diff
@@ -535,6 +541,76 @@ func (r *UserResolver) BatchChangesCodeHosts(ctx context.Context, args *ListBatc
...
+			if override.Value == true {
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/user.go:560]:
```suggestion
			if override.Value {
```

😁

**Review by unknwon — APPROVED** [review]
I think it would work, only reviewed backend part.

File: cmd/frontend/graphqlbackend/user.go (hunk 3 above)
Comment (miveronese) [reply cmd/frontend/graphqlbackend/user.go:559]: @unknwon  and @rafax , I am assuming that this PR will be merged before  mine ([the one that removes  the "github-app-cloud" flag from both back and frontend](https://github.com/sourcegraph/sourcegraph/pull/33370)).

Comment (unknwon) [reply cmd/frontend/graphqlbackend/user.go:559]: @miveronese My suggestion would be to not block merging of this PR from #33370, thus putting unnecessary extra time pressure on completing CLOUD-338. So yes, I'd assume the same.

Comment (rafax) [reply cmd/frontend/graphqlbackend/user.go:559]: Yes, feel free to merge yours and I'll update this one.

**Pull request closed without merging**